### PR TITLE
Use precision for samples only when #version 300 es

### LINKED ISSUE #10332: Vertex shader failed to compile [AMD, Windows 10, Regression from #9b9a723]
**Operating system or device - Godot version:**
Windows 10 (AMD FirePro D300); master https://github.com/godotengine/godot/commit/9575dbdf788e8a5154b3ec2f66913e731ac02850

**Issue description:**
Displays nothing but a white screen and following errors:
```
ERROR: _gl_debug_print: GL ERROR: Source: Shader Compiler	Type: Error	ID: 2000	Severity: High	Message: glCompileShader failed to compile a GLSL shader with the following shader info log: 'Vertex shader failed to compile with the following errors:
ERROR: 22:1: error(#228) Type should be float or int
ERROR: 23:1: error(#228) Type should be float or int
ERROR: 24:1: error(#228) Type should be float or int
ERROR: error(#273) 3 compilation errors.  No code generated
At: drivers\gles3\rasterizer_gles3.cpp:122
...
```

Default precision qualifiers for samplers (added in https://github.com/godotengine/godot/commit/9b9a723c771a0146cb16e23c3dac744af312bad7) only allowed in GLSL 4.40+, and Godot shaders are `#version 330`.

Replacing version with `#version 440` or removing `precision highp sampler*` lines fixes the problem, looks like AMD driver has too strict GLSL validation.

[GLSL 3.30 spec](https://www.khronos.org/registry/OpenGL/specs/gl/GLSLangSpec.3.30.pdf#page=49)
[GLSL 4.40 spec](https://www.khronos.org/registry/OpenGL/specs/gl/GLSLangSpec.4.40.pdf#page=95)

### PR DESCRIPTION
**Pull request by Faless**
Fix #10332

### CODE REVIEW

File: platform/javascript/detect.py [hunk 1]
```diff
@@ -112,6 +112,7 @@ def configure(env):
     else:
         env.Append(LINKFLAGS=['-s', 'ASM_JS=1'])
         env.Append(LINKFLAGS=['--separate-asm'])
+        env.Append(LINKFLAGS=['-s', 'FULL_ES2=1'])
```
Comment (eska014) [inline platform/javascript/detect.py]: This enables expensive GLES2 emulation, what does this fix?

**Comment by Faless** [comment]
I was afraid this would be problematic, but I can't seem to run the ASM.js
version at all without that option (getting a pointer error, I get a
undefined warning for glDrawArrays when linking).
Only the wasm version seems to work.

Does the HTML5 export with was=no works for you without FULL_ES2=1?


On Aug 14, 2017 20:18, "L. Krause" <notifications@github.com> wrote:

*@eska014* commented on this pull request.
------------------------------

In platform/javascript/detect.py
<https://github.com/godotengine/godot/pull/10343#discussion_r133022676>:

> @@ -112,6 +112,7 @@ def configure(env):
     else:
         env.Append(LINKFLAGS=['-s', 'ASM_JS=1'])
         env.Append(LINKFLAGS=['--separate-asm'])
+        env.Append(LINKFLAGS=['-s', 'FULL_ES2=1'])

This enables expensive GLES2 emulation, what does this fix?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/godotengine/godot/pull/10343#pullrequestreview-56159252>,
or mute the thread

**Comment by eska014** [comment]
Yeah, asm.js works as it is on my end. If WebAssembly works, I can't imagine what causes this. Maybe cache issues? Firefox uses a a separate cache for asm.js:
   > Note: to clear out the asm.js cache for a given origin, click the [Site Identity Button](https://support.mozilla.org/en-US/kb/how-do-i-tell-if-my-connection-is-secure) → More Information → Permissions → (scroll down) → Clear Storage.
   >https://blog.mozilla.org/luke/2014/01/14/asm-js-aot-compilation-and-startup-performance/

On other browsers Ctrl+F5 should be enough though.
Never seen a warning about `glDrawArrays` either, are you using an older Emscripten version for asm.js?

**Comment by Faless** [comment]
Oh, I see, must be something on my end then. I'll remove that as soon as I
get home

**Comment by reduz** [comment]
would be more interesting to know what is breaking for you first

**Comment by Faless** [comment]
I can't reproduce the error on my side anymore. It might have been caching a problem as @eska014 suggested.
I updated the PR

**Pull request merged**